packages/aws/src/RPA/Cloud/AWS/__init__.py

```diff
@@ -990,7 +1024,672 @@ def delete_queue(self, queue_name: str = None):
+          will be referenced as ``:name``.
+        * ``value``: The value of the parameter. Amazon Redshift implicitly
+          converts to the proper data type. For more inforation, see
+          `Data types`_ in the `Amazon Redshift Database Developer Guide`.
+
+        For simplicity, a helper keyword, \`Create redshift statement parameters\`,
+        is available and can be used more naturally in Robot Framework contexts.
+
+        .. _Data types: https://docs.aws.amazon.com/redshift/latest/dg/c_Supported_data_types.html
+
+        If tabular data is returned, this keyword tries to return it as
+        a table (see ``RPA.Tables``), if ``RPA.Tables`` is not available
+        in the keyword's scope, the data will be returned as a list of dictionaries.
+        Other types of data (SQL errors and result statements) are returned
+        as strings.
+
+        **NOTE:** You may modify the max built-in wait time by providing
+        a timeout in seconds (default 40 seconds)
+
+        **Robot framework example:**
+
+        .. code-block:: robotframework
+
+            *** Tasks ***
+
+                ${SQL}=    Set variable    insert into mytable values (:id, :address)
+                ${params}=    Create redshift statement parameters
+                ...    id=1
+                ...    address=Seattle
+                ${response}=    Execute redshift statement    ${SQL}    ${params}
+                Log    ${response}
+
+        **Python example:**
+
+        .. code-block:: python
+
+            sql = "insert into mytable values (:id, :address)"
+            parameters = [
+                {"name": "id", "value": "1"},
+                {"name": "address", "value": "Seattle"},
+            ]
+            response = aws.execute_redshift_statement(sql, parameters)
+            print(response)
+
+        :param parameters: The parameters for the SQL statement. Must consist
+            of a list of dictionaries with two keys: ``name`` and ``value``.
+        :param sql: The SQL statement text to run.
+        :param statement_name: The name of the SQL statement. You can name
+            the SQL statement when you create it to identify the query.
+        :param with_event: A value that indicates whether to send an event
+            to the Amazon EventBridge event bus after the SQL statement runs.
+        :param timeout: Used to calculate the maximum wait. Exact timing
+            depends on system variability becuase the underlying waiter
+            does not utilize a timeout directly.
+
+        """  # noqa: W605, E501
+        client = self._get_client_for_service("redshift-data")
+        run_token = self._submit_statement(
+            client, sql, parameters, statement_name, with_event
+        )
+        self.logger.info(
+            f"'{run_token['Id']}' SQL statement execution on Redshift started"
+            f"{' with name ' + statement_name if statement_name else ''}:\n{sql}"
```

Comment:
I find this `f""` formatting and its curly braces a little bit tricky here. What's the supposed outcome of formatting the `{ with name ...}` part? (I find hard to understand the intended behaviour)

Reply:
Broke the if statement to a line before to clarify behavior